### PR TITLE
League statistics

### PR DESCRIPTION
**Pull request by IamNorma**
Added league stats class methods:

- count_of_teams 
- best_offense 
- worst_offense 
- highest_scoring_visitor 
- highest_scoring_home_team 
- lowest_scoring_visitor 
- lowest_scoring_home_team

### CODE REVIEW

**Review by dborski — APPROVED** [review]

**Comment by dborski** [comment]
All the code looks super clean! Nicely done.

File: lib/league_stats.rb [hunk 1]
```diff
@@ -6,16 +6,79 @@ def initialize(game_teams, teams)
     @teams = teams
   end
 
-  def goals_by_team
-    games_by_team = @game_teams.group_by do |game|
-      game.team_id
+  def find_by_collection(team_id, csv_header, game_teams)
+    @game_teams.find_all { |team| team.send(csv_header) == team_id }
+  end
```
Comment (whitneykidd) [inline lib/league_stats.rb:11]: This method might be able to be utilized in other classes when we refactor too.

File: lib/league_stats.rb [hunk 2]
```diff
@@ -6,16 +6,79 @@ def initialize(game_teams, teams)
     @teams = teams
   end
 
-  def goals_by_team
-    games_by_team = @game_teams.group_by do |game|
-      game.team_id
+  def find_by_collection(team_id, csv_header, game_teams)
```
Comment (whitneykidd) [inline lib/league_stats.rb:9]: Looks good, we also might be able to utilize this method in the refactor 👍

**Comment by kiverso** [comment]
looks good, some potential for refactor with similar methods in our other classes down the line.

**Pull request merged**